SYCL/InvokeSimd/invoke_simd_struct.cpp

```diff
@@ -0,0 +1,168 @@
+// TODO: enable on Windows once driver is ready
+// REQUIRES: gpu && linux
+// UNSUPPORTED: cuda || hip
+//
+// Check that full compilation works:
+// RUN: %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr -I%S/../ESIMD/ %s -o %t.out
+// RUN: env IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+
+// Tests invoke_simd support in the compiler/headers
+
+/* Test case specification:
+ * -----------------------
+ * Test passing pointers to structs to invoke_simd functions.
+ *
+ * Test case description:
+ * -----------------------------
+ * This is a simple test case that defines a struct named
+ * multipliers with 2 int members, x and y. Suppose these
+ * members represent values by which we want to scale the
+ * addtion of vectors A + B. That is, we want to compute
+ * C = A + B * x * y (the sum of A + B scaled by x,
+ * and then scaled by y).
+ *
+ * Test case implementation notes:
+ * -------------------------------
+ * This test case has been modified from Konst's baseline
+ * test case, invoke_simd.cpp. It simply extends it by
+ * adding a struct pointer parameter to the invoke_simd
+ * functions which use it to scale the original vector
+ * addition.
+ */
```

Comment:
```suggestion
 * This test case is a modified
 * https://github.com/intel/llvm/blob/sycl/sycl/test/invoke_simd/invoke_simd.cpp.
 *  It simply extends it by
 * adding a struct pointer parameter to the invoke_simd
 * functions which use it to scale the original vector
 * addition.
 */
```

Reply:
One general note - please do not force-push, this makes it harder to see prior review comments/modifications.
If you want to get latest code from upstream, use `git checkout your_branch && git merge intel/intel`, then use simple `git push` when ready to push.